django/db/models/query.py

```diff
@@ -1575,6 +1575,17 @@ def prefetch_one_level(instances, prefetcher, lookup, level):
         instance_attr_val = instance_attr(obj)
         vals = rel_obj_cache.get(instance_attr_val, [])
         to_attr, as_attr = lookup.get_current_to_attr(level)
+
+        # Check we are not shadowing a field on obj
+        if as_attr:
+            try:
+                field = obj._meta.get_field(to_attr)
+            except exceptions.FieldDoesNotExist:
+                pass
+            else:
+                msg = 'to_attr={} shadows the {} field.'.format(to_attr, field)
```

Comment:
I'm not sure I'm a big fan of this error message. "Shadowing" has a specific meaning that may not be clear to everyone. Perhaps something along the lines of "You cannot use the name ... as the to_attr argument, as the model already has a field with this name"